scripts/earthengine/earthengine_image.py

```diff
@@ -649,29 +634,54 @@ def ee_process(config) -> list:
             if config.get('ee_export_image', True):
                 task = export_ee_image_to_gcs(img, config)
                 ee_tasks.append(task)
+                counters.add_counter('total', 1)
         else:
             logging.error(f'Failed to generate image for config: {config}')
         # Advance time to next period.
         time_period = config.get('time_period', 'P1M')
         for ts in ['start_date', 'end_date']:
-            dt = _advance_date(config.get(ts, ''), time_period)
+            dt = utils.date_advance_by_period(config.get(ts, ''), time_period)
             if dt:
                 config[ts] = dt
         config['ee_image_count'] = config['ee_image_count'] - 1
         logging.info(f'Advanced dates by {time_period}: {config}')
 
+    completed_tasks = []
+    if not config.get('ee_wait_task', True):
+        return ee_tasks
+
     # Wait for tasks to complete
-    if config.get('ee_wait_task', True):
-        pending_tasks = set(ee_tasks)
-        while len(pending_tasks) > 0:
-            task = pending_tasks.pop()
-            if task.active():
-                pending_tasks.add(task)
-                logging.info(f'Waiting for task: {task}')
-                time.sleep(10)
-            else:
-                task_status = ee.data.getTaskStatus(task.id)
-                logging.info(f'EE task completed: {task_status}')
+    pending_tasks = set(ee_tasks)
+    while len(pending_tasks) > 0:
+        task = pending_tasks.pop()
+        if task.active():
+            pending_tasks.add(task)
+            logging.info(f'Waiting for task: {task}')
+            time.sleep(10)
```

Comment:
nit: ideally, have the timeout (sleep) seconds be a constant somewhere so its easy to change?

Reply:
Done